### PR TITLE
Offset redundancy weighting

### PR DESCRIPTION
**Pull request by grupesh**
Added data redundancy weighting necessary for offset detector geometry (as an option) in all algorithms, except MLEM, tested with simulation & real data..
also includes 
- fix in V matrix for more stability
- additional fix in zeropadding function for geo.COR

### CODE REVIEW

**Comment by AnderBiguri** [comment]
Thanks a lot!

However, the optional parameter should come in as part of `varargin` as a keyword argument, similar to all the rest of the optional parameters. Can you please put the code there?  Let me know if this needs clarifying.

**Comment by grupesh** [comment]
Will do this in a bit

**Comment by grupesh** [comment]
changed redundancy weighting to be part of "varargin". Please review and merge

File: MATLAB/Algorithms/FDK.m [hunk 1]
```diff
@@ -148,7 +152,7 @@
 % Ref: 
 %    Wang, Ge. X-ray micro-CT with a displaced detector array. Medical Physics, 2002,29(7):1634-1636.
 offset = geo.offDetector(1);
-offset = offset + (geo.DSD(1)/geo.DSO(1))*geo.COR(1);
+offset = offset + (geo.DSD(1) / geo.DSO(1)) * geo.COR(1);   % added correction
```
Comment (AnderBiguri) [inline MATLAB/Algorithms/FDK.m:155]: We should remove `preweighting2()` totally and make `redundacy_weigting()` take the same arguments as `preweighting2()`. That way we don't have redundant code.

File: MATLAB/Algorithms/FDK.m [hunk 2]
```diff
@@ -247,7 +251,7 @@
             end
        
         otherwise
-            error('TIGRE:FDK:InvalidInput',['Invalid input name:', num2str(opt),'\n No such option in FDK()']);
```
Comment (AnderBiguri) [inline MATLAB/Algorithms/FDK.m:250]: They should all be `TIGRE` instead of `CBCT`

File: MATLAB/Utilities/computeV.m [hunk 3]
```diff
@@ -16,9 +17,9 @@
         auxindex=orig_index{ii};
         auxgeo = geo;
         % expand the detector to avoiding zeros in backprojection
-%         maxsize=max(auxgeo.sVoxel+geo.offOrigin(:,auxindex),[],2);
```
Comment (AnderBiguri) [inline MATLAB/Utilities/computeV.m:19]: These changes should not be in this PR.

**Review by AnderBiguri — CHANGES_REQUESTED** [review]

**Comment by AnderBiguri** [comment]
Hi @grupesh ,

Can I make commits to this branch? I am happy to tackle the reviews myself to add it to TIGRE.

**Comment by grupesh** [comment]
Sure sure, sorry about the delays on my end..

**Comment by AnderBiguri** [comment]
@grupesh  absolutely no worries! This is a free contribution to an open source software! Thanks a lot !

**Comment by AnderBiguri** [comment]
Final question/comment: in the wang weigthts in FDK, they are multiplied by 2 in the end. Is there any reason you have not done this in your `reundacy_weighting()` function? just forgot, or is it better without it?

**Comment by grupesh** [comment]
I've kept multiplication by 2 in redundancy weighting, same as
preweighting2. I think multiplication by 2 is helpful/necessary, in the
definition of weights in Wang paper, multiplication factor of 2 is present.

Best,
Rupesh

**Comment by AnderBiguri** [comment]
Where is the multiplicatioin happening? I can't see it

**Comment by AnderBiguri** [comment]
I think this is ready to merge. If you think so too, I will!

Thanks again for the great work.

**Comment by grupesh** [comment]
Yes, I had forgotten the multiplication by 2 factor, thanks for taking care of that and cleaning up the code. 
I think it's ready to merge.

**Comment by AnderBiguri** [comment]
Thanks to you for the hard work @grupesh

**Pull request merged**